src/Altinn.Platform/Altinn.Platform.Storage/Storage/Controllers/InstancesController.cs

```diff
@@ -323,16 +345,152 @@ public async Task<ActionResult> Post(string appId, int? instanceOwnerId, [FromBo
             try
             {
                 Instance result = await _instanceRepository.Create(createdInstance);
-                return Ok(result);
+
+                if (result == null)
+                {
+                    throw new ArgumentNullException("result");
```

Comment:
A better error description would be: Failed to create instance in repository

Reply:
Removed the null check and ArgumentNullException, since the Create method in instance repository will throw error itself.